src/Enzo/enzo_EnzoConstrainedTransport.cpp

```diff
@@ -0,0 +1,467 @@
+// See LICENSE_CELLO file for license and copyright information
+
+/// @file     enzo_EnzoConstrainedTransport.cpp
+/// @author   Matthew Abruzzo (matthewabruzzo@gmail.com)
+/// @date     Mon May 6 2019
+/// @brief    [\ref Enzo] Implementation of EnzoConstrainedTransport
+
+#include "cello.hpp"
+#include "enzo.hpp"
+
+//----------------------------------------------------------------------
+
+void EnzoConstrainedTransport::compute_center_efield
+(Block *block, int dim, std::string center_efield_name, Grouping &prim_group,
+ int stale_depth)
+{
+  // Load the E-field
+  EnzoFieldArrayFactory array_factory(block,stale_depth);
+  EFlt3DArray efield = array_factory.from_name(center_efield_name);
+
+  EnzoPermutedCoordinates coord(dim);
+  int j = coord.j_axis();
+  int k = coord.k_axis();
+
+  // Load the jth and kth components of the velocity and cell-centered bfield
+  EFlt3DArray velocity_j, velocity_k, bfield_j, bfield_k;
+  velocity_j = array_factory.from_grouping(prim_group, "velocity", j);
+  velocity_k = array_factory.from_grouping(prim_group, "velocity", k);
+  bfield_j = array_factory.from_grouping(prim_group, "bfield", j);
+  bfield_k = array_factory.from_grouping(prim_group, "bfield", k);
+
+  for (int iz=0; iz<efield.shape(0); iz++) {
+    for (int iy=0; iy<efield.shape(1); iy++) {
+      for (int ix=0; ix<efield.shape(2); ix++) {
```

Comment:
Everywhere in these new additions , except in the methods of `CelloArray` we have the following mapping:
- `0 <--> x`
- `1 <--> y`
- `2 <--> z`

I do agree that it would probably be worth addressing in the `CelloArray` (probably by making the dimension numbering similar to Athena Arrays). 

To make things more explicit, we could probably try using enums. That's an idea I had considered, but never really pursued.